rfd/0227-ec2-discovery-by-aws-org.md

```diff
@@ -0,0 +1,442 @@
+
+  Policy:
+  ```json
+    {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Effect": "Allow",
+                "Action": [
+                    "organizations:ListAccounts",
+                    "sts:AssumeRole"
+                ],
+                "Resource": [
+                    "*"
+                ]
+            }
+        ]
+    }
+  ```
+  
+  Trust Policy:
+  ```json
+    {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Effect": "Allow",
+                "Principal": {
+                    "Service": "ec2.amazonaws.com"
+                },
+                "Action": "sts:AssumeRole"
+            }
+        ]
+    }
+  ```
+</details>
+
+The user also needs to create an IAM Role in each Account, which will be assumed by the Role assigned to the Discovery Service (Role above):
+
+<details>
+  <summary>IAM Role to create in every Account</summary>
+
+  Policy:
+  ```json
+    {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Effect": "Allow",
+                "Action": [
+                    "ec2:DescribeInstances",
+                    "ssm:DescribeInstanceInformation",
+                    "ssm:GetCommandInvocation",
+                    "ssm:ListCommandInvocations",
+                    "ssm:SendCommand"
+                ],
+                "Resource": [
+                    "*"
+                ]
+            }
+        ]
+    }
+  ```
+  
+  Trust Policy:
+  ```json
+    {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Effect": "Allow",
+                "Principal": {
+                    "AWS": "<Role's ARN assigned to the Discovery Service>"
+                },
+                "Action": "sts:AssumeRole"
+            }
+        ]
+    }
+  ```
+</details>
+
+Still in step 2, the Alice is asked to add the `AmazonSSMManagedInstanceCore` IAM managed policy to the target EC2 instances.
+Alongside this policy, the user must also include the `organizations:DescribeOrganization` action:
+<details>
+  <summary>IAM Role to create in each Account</summary>
+
+  Policy:
+  ```json
+    {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Effect": "Allow",
+                "Action": [
+                    "organizations:DescribeOrganization"
+                ],
+                "Resource": [
+                    "*"
+                ]
+            }
+        ]
+    }
+  ```
+  
+  Trust Policy:
+  ```json
+    {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Effect": "Allow",
+                "Principal": {
+                    "Service": "ec2.amazonaws.com"
+                },
+                "Action": "sts:AssumeRole"
+            }
+        ]
+    }
+  ```
+</details>
+
+Going to step 3, Alice is now asked to create an SSM Document in every Account and Region that have EC2 instances.
```

Comment:
As previously discussed, there may be a way we can use default SSM document to remove this step? Would make setup a bit simpler.

Reply:
Honestly, this is the most boring task: you have to create a document per account and per region.

I'll add to the RFD that we will remove this step and use the `AWS-RunShellScript` managed SSM Document.